core/csv_logger.py

```diff
@@ -0,0 +1,127 @@
+import csv
+import re
+import threading
+import time
+from pathlib import Path
+from typing import TYPE_CHECKING, Union, Mapping
+
+_stdlock = threading.Lock()
+_filelock = threading.Lock()
+
+
+if TYPE_CHECKING:
+    from scm.plams.core.basejob import Job
+
+
+class LoggerCSV:
+    """basic logger that mimic the behavior of plams.log with adding option of personalize the file logging location"""
+
+    def __init__(self, log_path="logfile.csv", stdout_level=3, date=True, time=True, init: bool = True):
+        self.log_path = Path(log_path)
+        self.stdout_level = stdout_level
+        self.date = date
+        self.time = time
+        self.init = init
+
+    def log(self, message: Mapping[str, Union[str, int, float]], level=7):
+        """Log a dictionary message in CSV format with verbosity level.
+
+        Logs are printed independently to the CSV logfile and to the standard output based on configuration levels.
+        Date and/or time can be added based on configuration. All logging activity is/should thread safe.
+
+        :param message: The message to log as a dictionary.
+        :type message: dict
+        :param level: 1: important; 3: normal; 5: verbose; 7: debug, defaults to 0
+        :type level: int, optional
+        """
+        if not self.init:
```

Comment:
Is this needed here? For the main `log` function, it is checking whether init has been called (and therefore the default job manager set up) but here we don't need to wait for this, or anything else. The path has been set in the `__init__` function

Reply:
true. It is a remnant due to historical reasons (at the beginning `LoggerCSV` was a function like the `log` one).